google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -328,25 +329,29 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
   }
 
   @Test
-  public void testResponseHeaderContainsMetaDataFlavor() throws Exception {  
-    Map<String, String> headers = new HashMap<String, String>();
-    HttpResponse httpResponse = createHttpResponseWithHeader(headers);
-    assertThat(ServiceOptions.headerContainsMetadataFlavor(httpResponse)).isFalse();
-
+  public void testResponseHeaderContainsMetaDataFlavor() throws Exception {
+    Multimap<String, String> headers = ArrayListMultimap.create();
     headers.put("Metadata-Flavor", "Google");
-    httpResponse = createHttpResponseWithHeader(headers);
+    HttpResponse httpResponse = createHttpResponseWithHeader(headers);
     assertThat(ServiceOptions.headerContainsMetadataFlavor(httpResponse)).isTrue();
   }
+
+  @Test
+  public void testResponseHeaderDoesNotContainMetaDataFlavor() throws Exception {  
+    Multimap<String, String> headers = ArrayListMultimap.create();
+    HttpResponse httpResponse = createHttpResponseWithHeader(headers);
+    assertThat(ServiceOptions.headerContainsMetadataFlavor(httpResponse)).isFalse(); 
+  }
   
-  private HttpResponse createHttpResponseWithHeader(final Map<String, String> headers) throws Exception {
+  private HttpResponse createHttpResponseWithHeader(final Multimap<String, String> headers) throws Exception {
     HttpTransport mockHttpTransport = new MockHttpTransport() {
       @Override
       public LowLevelHttpRequest buildRequest(String method, String url) throws IOException {
         return new MockLowLevelHttpRequest() {
           @Override
           public LowLevelHttpResponse execute() throws IOException {
             MockLowLevelHttpResponse response = new MockLowLevelHttpResponse();
-            for (Map.Entry<String, String> entry : headers.entrySet()) {
+            for (Map.Entry<String, String> entry : headers.entries()) {
```

Comment:
Just as an FYI--it's not like it makes a difference here, but just so you know :)--this simplifies to one line using streams:

`headers.entries().forEach(e -> response.addHeader(e.getKey(), e.getValue());`

Reply:
We need Java 7 compat, so unfortunately so streams for us :(

Reply:
:(
Will this change once the App Engine Standard Java 7 runtime is decommissioned?